modules/currency-l0/src/main/scala/org/tessellation/currency/l0/modules/Daemons.scala

```diff
@@ -22,14 +23,18 @@ object Daemons {
     programs: Programs[F],
     queues: Queues[F],
     healthChecks: HealthChecks[F],
-    maybeDataApplication: Option[BaseDataApplicationL0Service[F]]
-  ): F[Unit] =
+    maybeDataApplication: Option[BaseDataApplicationL0Service[F]],
+    config: AppConfig
+  ): F[Unit] = {
+    val downloadDaemon = DelayForPeers.make(storages.cluster, config.environment)
+
     List[Daemon[F]](
       NodeStateDaemon.make(storages.node, services.gossip),
-      DownloadDaemon.make(storages.node, programs.download),
+      DownloadDaemon.make(storages.node, programs.download, downloadDaemon),
```

Comment:
Why "downloadDaemon" takes "downloadDaemon" as an argument?

Reply:
Oversight. The variable has been renamed to `downloadDelay`.